### DISCUSSION TITLE
How to find what causes being stucked at SSR bundle building?

### DISCUSSION THREAD

**Discussion by ayZagen** [antfu/vite-ssg#260] [Q&A]
Hey there,

I have a vite-ssg project using `vite-plugin-pages`, `vite-plugin-pages-sitemap` and `vite-plugin-pwa`. It runs OK in dev but when I try to get a production build it stucks at the stage of `building SSR bundle for production... transforming (1) src\main.ts`

It seems like a circular dependency issue (?) but I could not find the culprit. 

I tried to remove pages/files one by one to find which file is causing the loop but couldn't made a progress. Also tried using [madge](https://github.com/pahen/madge) to detect circular dependencies... with no luck. 

Here is the output:
```shell
> cross-env DEBUG=vite-ssg:* vite-ssg build

Running vite in "production" mode

[vite-ssg] Build for client...
Running vite in "production" mode
vite v2.9.12 building for production...
✓ 1123 modules transformed.
dist/assets/....                     xxx KiB
/****** Reducted for the log simplicity  ******/

PWA v0.12.0
mode      generateSW
precache  64 entries (1059.42 KiB)
files generated
  dist\sw.js
  dist\workbox-f51ab5e4.js

[vite-ssg] Build for server...
Running vite in "production" mode
vite v2.9.12 building SSR bundle for production...
transforming (1) src\main.ts
```

What can i do to find what is causing the problem?

**Comment by userquin** [comment]
@ayZagen maybe adding `rollupOptions`  under `build`?:
```ts
    rollupOptions: {
      onwarn(message) {
        // or just log everything
        if (/Circular dependencies/.test(message.message))
          console.error(message)
      },
    },
```

**Comment by ayZagen** [comment]
It was happening because of `unplugin-auto-import/vite` and `unplugin-vue-components/vite` I removed them and did a LOT of refactoring now project builds fine.
